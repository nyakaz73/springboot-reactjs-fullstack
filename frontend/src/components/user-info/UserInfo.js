import React from 'react'
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import CreateIcon from '@material-ui/icons/Create';
import PropTypes from "prop-types"
const UserInfo = ({ user, removeUser }) => {

    const { id, name, surname, email, username } = user;
    const infoStyle = {
        backgroundColor: (id % 2) == 0 ? '#c8d4f7cc' : '',
        padding: '10px',
        borderBottom: '1px #ccc dotted',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        justifyItems: 'flex-start',
        marginLeft: '20px',
        marginRight: '20px',


    }

    const deleteUser = (id) => {
        removeUser(id);
    }


    return (
        <div style={infoStyle}>
            <div style={iconUsername}>
                <AccountCircleIcon style={{ color: "#138a04", margin: "0px 10px 0px 0px" }} />
                <p>{name}</p>
                &nbsp;
                <p>{surname}</p>
            </div>

            <p>{email}</p>
            <p>{username}</p>
            <div style={buttons}>
                {/*<IconButton color="secondary" onClick={this.props.removeUser.bind(this,id )} >  Because binding in faling in Jest i have used arrow function binding*/}
                <IconButton style={{ color: '#888990' }}  >
                    <CreateIcon />
                </IconButton>
                <IconButton aria-label="delete-button" color="secondary" onClick={(e) => deleteUser(id)} >
                    <DeleteIcon />
                </IconButton>
            </div>
        </div>
    )
}

const iconUsername = {
    display: 'flex',
    justifyContent: 'space-between',
}

const buttons = {
    display: 'flex'
}

UserInfo.propTypes = {
    user: PropTypes.object,
    removeUser: PropTypes.func.isRequired
}
export default UserInfo;
