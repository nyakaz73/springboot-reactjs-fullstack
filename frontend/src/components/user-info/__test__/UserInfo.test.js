import React from "react";
import ReactDOM from "react-dom";
import { act, render } from "@testing-library/react";
import UserInfo from "../UserInfo";
import renderer from "react-test-renderer"

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it("It Renders <UserInfo/> without crashing", () => {
    const removeUser = jest.fn()
    const user = {
        'name': 'John',
        'surname': 'Doe',
        'email': 'john@gmail.com',
        'username': 'johnny',
    }
    act(() => {
        ReactDOM.render(<UserInfo user={user} removeUser={removeUser} />, container);
    });
    expect(container.textContent).toContain(user.name);
    expect(container.textContent).toContain(user.surname);
    expect(container.textContent).toContain(user.email);
    expect(container.textContent).toContain(user.username);

})

it("onClick deleteUser button fired", () => {
    const removeUser = jest.fn()
    const user = {
        'name': 'John',
        'surname': 'Doe',
        'email': 'john@gmail.com',
        'username': 'johnny',
    }

    const utils = render(<UserInfo user={user} removeUser={removeUser} />)
    const removeButton = utils.getByLabelText('delete-button')
    act(() => {
        removeButton.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    //removeButton.dispatchEvent(new MouseEvent('click'));
    expect(removeUser).toHaveBeenCalledTimes(1)
})

it("<UserInfo/> matches snapshot", () => {
    const removeUser = jest.fn()
    const user = {}
    const tree = renderer.create(<UserInfo user={user} removeUser={removeUser} />).toJSON();
    expect(tree).toMatchSnapshot();
})

