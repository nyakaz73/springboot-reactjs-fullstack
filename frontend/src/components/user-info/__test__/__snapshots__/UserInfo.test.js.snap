// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo/> matches snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "",
      "borderBottom": "1px #ccc dotted",
      "display": "flex",
      "justifyContent": "space-between",
      "justifyItems": "flex-start",
      "marginLeft": "20px",
      "marginRight": "20px",
      "padding": "10px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "color": "#138a04",
          "margin": "0px 10px 0px 0px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <p />
    Â 
    <p />
  </div>
  <p />
  <p />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "color": "#888990",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </span>
    </button>
    <button
      aria-label="delete-button"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
